dali/dfuplus/dfuplus.cpp

```diff
@@ -192,6 +193,15 @@ CDfuPlusHelper::CDfuPlusHelper(IProperties* _globals,   CDfuPlusMessagerIntercep
 
     const char* username = globals->queryProp("username");
     const char* password = globals->queryProp("password");
+    if ( username && *username && (!password || !*password))
+    {
+        VStringBuffer prompt("%s's password: ", username);
+        StringBuffer passw;
+        passw = passwordInput(prompt, passw);
+        globals->setProp("password",passw.str());
+        password = globals->queryProp("password");
```

Comment:
Yes, I see that, I guess I would usually create passw outside the if and only use it if needed.  But ultimately mostly a style difference I guess.  Avoid creating a potentially empty StringBuffer or avoid a hash lookup.